File: samples/openapi3/server/petstore/spring-boot-oneof-sealed/src/main/java/org/openapitools/model/Cat.java
package org.openapitools.model;

import java.net.URI;
import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import org.springframework.lang.Nullable;
import org.openapitools.jackson.nullable.JsonNullable;
import java.time.OffsetDateTime;
import javax.validation.Valid;
import javax.validation.constraints.*;
import io.swagger.v3.oas.annotations.media.Schema;


import java.util.*;
import javax.annotation.Generated;

/**
 * Cat
 */

@Generated(value = "org.openapitools.codegen.languages.SpringCodegen", comments = "Generator version: 7.17.0-SNAPSHOT")
public final class Cat implements Animal {

  private @Nullable Boolean declawed;

  public Cat declawed(@Nullable Boolean declawed) {
    this.declawed = declawed;
    return this;
  }

  /**
   * Get declawed
   * @return declawed
   */
  
  @Schema(name = "declawed", requiredMode = Schema.RequiredMode.NOT_REQUIRED)
  @JsonProperty("declawed")
  public @Nullable Boolean getDeclawed() {
    return declawed;
  }

  public void setDeclawed(@Nullable Boolean declawed) {
    this.declawed = declawed;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    Cat cat = (Cat) o;
    return Objects.equals(this.declawed, cat.declawed);
  }

  @Override
  public int hashCode() {
    return Objects.hash(declawed);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class Cat {\n");
    sb.append("    declawed: ").append(toIndentedString(declawed)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
  
  public static class Builder {

    private Cat instance;

    public Builder() {
      this(new Cat());
    }

    protected Builder(Cat instance) {
      this.instance = instance;
    }

    protected Builder copyOf(Cat value) { 
      this.instance.setDeclawed(value.declawed);
      return this;
    }

    public Cat.Builder declawed(Boolean declawed) {
      this.instance.declawed(declawed);
      return this;
    }
    
    /**
    * returns a built Cat instance.
    *
    * The builder is not reusable (NullPointerException)
    */
    public Cat build() {
      try {
        return this.instance;
      } finally {
        // ensure that this.instance is not reused
        this.instance = null;
      }
    }

    @Override
    public String toString() {
      return getClass() + "=(" + instance + ")";
    }
  }

  /**
  * Create a builder with no initialized field (except for the default values).
  */
  public static Cat.Builder builder() {
    return new Cat.Builder();
  }

  /**
  * Create a builder with a shallow copy of this instance.
  */
  public Cat.Builder toBuilder() {
    Cat.Builder builder = new Cat.Builder();
    return builder.copyOf(this);
  }

}

File: samples/server/petstore/kotlin-springboot-include-http-request-context-delegate/src/main/kotlin/org/openapitools/model/Cat.kt
package org.openapitools.model

import java.util.Objects
import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.annotation.JsonValue
import org.openapitools.model.Category
import org.openapitools.model.Color
import org.openapitools.model.Pet
import org.openapitools.model.Tag
import javax.validation.constraints.DecimalMax
import javax.validation.constraints.DecimalMin
import javax.validation.constraints.Email
import javax.validation.constraints.Max
import javax.validation.constraints.Min
import javax.validation.constraints.NotNull
import javax.validation.constraints.Pattern
import javax.validation.constraints.Size
import javax.validation.Valid
import io.swagger.annotations.ApiModelProperty

/**
 * 
 * @param hunts 
 * @param age 
 */
data class Cat(

    @ApiModelProperty(example = "null", required = true, value = "")
    @get:JsonProperty("name", required = true) override val name: kotlin.String,

    @ApiModelProperty(example = "null", required = true, value = "")
    @get:JsonProperty("photoUrls", required = true) override val photoUrls: kotlin.collections.List<kotlin.String>,

    @ApiModelProperty(example = "null", required = true, value = "")
    @get:JsonProperty("petType", required = true) override val petType: kotlin.String,

    @ApiModelProperty(example = "null", value = "")
    @get:JsonProperty("hunts") val hunts: kotlin.Boolean? = null,

    @ApiModelProperty(example = "null", value = "")
    @get:JsonProperty("age") val age: kotlin.Int? = null,

    @ApiModelProperty(example = "null", value = "")
    @get:JsonProperty("id") override val id: kotlin.Long? = null,

    @field:Valid
    @ApiModelProperty(example = "null", value = "")
    @get:JsonProperty("category") override val category: Category? = null,

    @field:Valid
    @ApiModelProperty(example = "null", value = "")
    @get:JsonProperty("tags") override val tags: kotlin.collections.List<Tag>? = null,

    @field:Valid
    @ApiModelProperty(example = "null", value = "")
    @get:JsonProperty("color") override val color: Color? = null
) : Pet, java.io.Serializable {

    companion object {
        private const val serialVersionUID: kotlin.Long = 1
    }
}

